lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -648,18 +692,27 @@ export class UserAgentApplication {
         return;
       }
 
-      //if user is not currently logged in and no login_hint is passed
-      if (!userObject && !(extraQueryParameters && (extraQueryParameters.indexOf(Constants.login_hint) !== -1))) {
+      // If no session exists, prompt the user to login.
+      if (!userObject && !!(request.sid  || request.loginHint)) {
         this.logger.info("User login is required");
         reject(ErrorCodes.userLoginError + Constants.resourceDelimiter + ErrorDescription.userLoginError);
         return;
       }
 
+      // construct extraQueryParams string from the request
+      let extraQueryParameters = Utils.constructExtraQueryParametersString(request.extraQueryParameters);
+      extraQueryParameters = Utils.addPromptParameter(extraQueryParameters, request.prompt);
+
+      // if the developer provides one of these, give preference to developer choice
+      if (request.account || request.sid || request.loginHint) {
+        extraQueryParameters = Utils.constructUnifiedCacheExtraQueryParameter(extraQueryParameters, request, null);
+      }
+
       // track the acquireToken progress
       this.acquireTokenInProgress = true;
 
       let authenticationRequest: AuthenticationRequestParameters;
-      const acquireTokenAuthority = authority ? AuthorityFactory.CreateInstance(authority, this.config.auth.validateAuthority) : this.authorityInstance;
+      const acquireTokenAuthority = request.authority ? AuthorityFactory.CreateInstance(request.authority, this.config.auth.validateAuthority) : this.authorityInstance;
```

Comment:
should this be this.authorityInstance || AuthorityFactory.create ...
is there ever a chance we have an authority instance but no request.authority?

Reply:
double check logic

Reply:
As explained offline, this is needed. We can set the authority per request and that is what the CreateInstance() call here does. In case of no authority specified in a request, we default to what is set in the configuration.